src/library/searchquery.cpp

```diff
@@ -166,6 +173,38 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument,
+                                 const CrateStorage* pCrateStorage)
+    : m_database(database),
+      m_argument(argument),
+      m_pCrateStorage(pCrateStorage) {
+    init();
+}
+
+void CrateFilterNode::init() {
+    CrateTrackSelectResult crateTracks(
+        m_pCrateStorage->selectTracksSortedByCrateNameLike(m_argument));
+
+    while (crateTracks.next()) {
+        m_trackIds.push_back(crateTracks.trackId());
+    }
+}
+
+bool CrateFilterNode::match(const TrackPointer& pTrack) const {
+    return std::binary_search(m_trackIds.begin(), m_trackIds.end(), pTrack->getId());
+}
+
+QString CrateFilterNode::toSql() const {
```

Comment:
This code simply doesn't scale. For example I have crates with thousands of tracks that I use for storing pre-computed searches.
Subselect: "id IN (" + CrateStorage::formatSubselectQueryForCrateTrackIdsWhereCrateNameLike(crateNameLike) + ")"

Please note, that you need to escape crateNameLike manually when formatting the subselect string! We don't have access to the SQL query of the caller and can't bind the parameter value by name here.

I also came to the conclusion that we might use "Where" instead of "By" for the filtering condition in those function names. Terms like "...SortedBy..." are simply misleading.